code/datums/components/footstep.dm

```diff
@@ -2,33 +2,56 @@
 	var/steps = 0
 	var/volume
 	var/e_range
-
-/datum/component/footstep/Initialize(volume_ = 0.5, e_range_ = -1)
+	var/footstep_type
+	///This can be a list OR a soundfile OR null. Determines whatever sound gets played.
+	var/footstep_sounds
+
+///Explanation of the args.
+///volume determines the extra volume of the footstep. This is multiplied by the base volume, should there be one.
+///e_range stands for extra range - aka how far the sound can be heard. This is added to the base value and ignored if there isn't one.
+///footstep_type is a define which determines what kind of sounds should get chosen.
+/datum/component/footstep/Initialize(footstep_type_ = FOOTSTEP_MOB_BAREFOOT, volume_ = 0.5, e_range_ = -1)
 	if(!isliving(parent))
 		return COMPONENT_INCOMPATIBLE
 	volume = volume_
 	e_range = e_range_
-	RegisterSignal(parent, list(COMSIG_MOVABLE_MOVED), .proc/play_footstep)
-
-/datum/component/footstep/proc/play_footstep()
+	footstep_type = footstep_type_
+	switch(footstep_type)
+		if(FOOTSTEP_MOB_HUMAN)
+			if(!ishuman(parent))
+				return COMPONENT_INCOMPATIBLE
+			RegisterSignal(parent, list(COMSIG_MOVABLE_MOVED), .proc/play_humanstep)
+			return
+		if(FOOTSTEP_MOB_CLAW)
+			footstep_sounds = GLOB.clawfootstep
+		if(FOOTSTEP_MOB_BAREFOOT)
+			footstep_sounds = GLOB.barefootstep
+		if(FOOTSTEP_MOB_HEAVY)
+			footstep_sounds = GLOB.heavyfootstep
+		if(FOOTSTEP_MOB_SHOE)
+			footstep_sounds = GLOB.footstep
+		if(FOOTSTEP_MOB_SLIME)
+			footstep_sounds = 'sound/effects/footstep/slime1.ogg'
+	RegisterSignal(parent, list(COMSIG_MOVABLE_MOVED), .proc/play_simplestep) //Note that this doesn't get called for humans.
+
+///Prepares a footstep. Determines if it should get played. Returns the turf it should get played on. Note that it is always a /turf/open
+/datum/component/footstep/proc/prepare_step()
 	var/turf/open/T = get_turf(parent)
 	if(!istype(T))
```

Comment:
You dont need to istype, get_turf always returns a turf or null
```suggestion
	if(!T)
```

Reply:
Footsteps don't exist on non open turfs.